qlib/kernel/loader/vdso.rs

```diff
@@ -39,6 +42,83 @@ pub struct VdsoInternal {
     pub vdsoAddr: u64,
     pub vdsoLen: usize,
     pub phdrs: Vec<ProgramHeader64>,
+    pub vdso_symbols: Option<VdsoArchSymbols>,
+}
+
+struct VdsoSymbol {
+    name:&'static str,
+    page_offset: Option<u64>,
+}
+
+/// Bookkeep the exported symbols for x86_64.
+#[cfg(target_arch = "x86_64")]
+pub struct VdsoArchSymbols {
+}
+
+#[cfg(target_arch = "x86_64")]
+impl VdsoArchSymbols {
+    pub fn set_symbols_page_offset(&mut self, elf_file: &ElfFile, symbol_table: &[Entry64]) {
+        todo!();
+    }
+
+    pub fn get_symbol_page_offset(&self, name: &str) -> Option<u64> {
+       todo!();
+       None
+    }
+}
+
+/// Bookkeep exported symbols for aarch64.
+#[cfg(target_arch = "aarch64")]
+pub struct VdsoArchSymbols {
+    symbols:[VdsoSymbol; Self::TOTAL_SYMBOLS],
+}
+
+#[cfg(target_arch = "aarch64")]
+impl Default for VdsoArchSymbols {
+    fn default() -> Self {
+        let _symbols: [VdsoSymbol; Self::TOTAL_SYMBOLS] = [
+            VdsoSymbol {name: "__kernel_gettimeofday", page_offset: None},
+            VdsoSymbol {name: "__kernel_clock_getres", page_offset: None},
+            VdsoSymbol {name: "__kernel_rt_sigreturn", page_offset: None},
```

Comment:
For completeness yes, but I don't think we will ever use "vsyscall" function symbols other than the sigreturn in kernel mode.

Reply:
I want to adhere to the documentation (as it is minimal).